### ISSUE TITLE
Fullschreen mode stacking order setting

### ISSUE THREAD

**Issue by NicTanghe** [leftwm/leftwm#943]
### Describe the enhancement

My main applications spawn allot of windows where i need to configure things.

Currently the fullshreen window always tries to stay on top which is incredibly annoying.
It whould be great if there where options to change the stacking order for fullschreen windows, preferably in the window_rules.

Preferably something like
```rust
    window_rules: [
        ( window_class: "Houdini FX", spawn_floating:  true , fs_stackorder : back),
    ],
```

### How important is this enhancement to you?

My workflow is impaired without it

### How important do you think is this enhancement for the average LeftWM user?

Their workflow is likely impaired without it

**Comment by NicTanghe** [comment]
Hello just a quick bump here.

**Comment by BKSalman** [comment]
Was this solved by #1121 ?

**Comment by BKSalman** [comment]
Was this solved by #1121 ? @NicTanghe

**Comment by Eskaan** [comment]
No, the maximized windows have the same stacking order as fullscreen ones, so they are above anything.

**Comment by BKSalman** [comment]
> No, the maximized windows have the same stacking order as fullscreen ones, so they are above anything.

Nope, I made floating windows go on top of em (if not then that's a bug in my code)

**Comment by Eskaan** [comment]
I don't experience that on your branch

**Comment by BKSalman** [comment]
welp, seems like yet another thing I missed in my PR (even tho I remember handling it, but I think I missed it somehow)

**Comment by BKSalman** [comment]
Ok seems like I just didn't implement it well, then broke the code even more, but now I will submit a PR that fixes it.

@Eskaan could you please review it to make sure I'm not making another mistake?

**Comment by Eskaan** [comment]
Review what exactly?

**Comment by VuiMuich** [comment]
I believe @BKSalman is referring to #1134 right?

**Comment by Eskaan** [comment]
> I don't experience that on your branch

#1134 is what I am referring to here. I was just told on Discord that the pr he was referring to isn't submitted yet.

**Comment by NicTanghe** [comment]
Hello,
I`m not quite sure that I`d classify this as fixed.
I get the feeling that it is allot better than what it used to be.
The full-screen window now consistently changes to a tiled window with the floating windows on top of it. 

So the window changes to a tiled window instead of a sunscreen one, which honestly isn’t that bad.

**Comment by BKSalman** [comment]
@NicTanghe do you mean maximized?

**Comment by BKSalman** [comment]
@NicTanghe um, I don't think you're talking about fullscreen (?), since it fullscreen doesn't even render any other window (not even the dock), so it can't show up, maybe you're talking about maximized windows?

if I'm wrong please correct me @VuiMuich @Eskaan

**Comment by NicTanghe** [comment]
wel the fulschreen window stops being a fullshcreen window when it spawns windows.

ps. not sre if it is relevant but I`m having most problems with a wind with these xwindow props.

```
❯ xprop
_NET_WM_DESKTOP(CARDINAL) = 4
WM_STATE(WM_STATE):
		window state: Normal
		icon window: <field not available>
_NET_WM_STATE(ATOM) =
_NET_WM_USER_TIME(CARDINAL) = 12282748
WM_TRANSIENT_FOR(WINDOW): window id # 0x5a0003c
_NET_WM_ICON(CARDINAL) =
_NET_WM_ICON_NAME(UTF8_STRING) =
XdndAware(ATOM) = BITMAP
WM_NAME(STRING) = "infini"
_NET_WM_NAME(UTF8_STRING) = "infini"
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x3, 0x26, 0x1e, 0x0, 0x0
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_UTILITY, _NET_WM_WINDOW_TYPE_NORMAL
_XEMBED_INFO(_XEMBED_INFO) = 0x0, 0x1
WM_CLIENT_LEADER(WINDOW): window id # 0x5a0000b
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: True
		window id # of group leader: 0x5a0000b
WM_CLIENT_MACHINE(STRING) = "RHEL-FX-workstation-01"
_NET_WM_PID(CARDINAL) = 88278
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 94372092
WM_CLASS(STRING) = "Houdini FX", "Houdini FX"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		user specified location: 2212, 180
		user specified size: 519 by 890
		program specified minimum size: 195 by 647
		window gravity: Static
```

**Comment by Eskaan** [comment]
I want to note that any newly spawned window forces all windows on that tag to exit fullscreen.

I would assume there to be some code that explicitly does this because until [a few commits ago](020ce3ae8f407165a084aa52501e8bf45f6ca2f1), no window could be above a window in fullscreen.

**Comment by NicTanghe** [comment]
It seems the fullschreen window no longer tries to stay on top after going into windowed mode.
I`ll test with a maximised window next.
